functions/slaViolationCloseScheduling/src/app/lib/sqs.js

```diff
@@ -0,0 +1,85 @@
+const { SQSClient, SendMessageCommand } = require("@aws-sdk/client-sqs");
+
+const client = new SQSClient({ region: process.env.REGION });
+
+exports.addActiveSLAToQueue = async (violations) => {
+  const response = {
+    receivedViolations: 0,
+    correctlySentViolations: 0,
+    problemsSendingViolations: 0,
+    skippedViolations: 0,
+  };
+
+  if (violations == null || !Array.isArray(violations)) {
+    return response;
+  }
+
+  response.receivedViolations = violations.length;
+
+  for (const singleViolation of violations) {
+    if (
+      singleViolation.entityName_type_relatedEntityId === undefined ||
+      singleViolation.id === undefined
+    ) {
+      response.skippedViolations++;
+      continue; // we skip this violation
+    }
+    let body = "";
+    try {
+      body = JSON.stringify(singleViolation);
+    } catch (error) {
+      /* istanbul ignore next */
+      response.skippedViolations++;
+      /* istanbul ignore next */
+      continue;
+    }
+    const params = {
+      DelaySeconds: 10,
+      MessageBody: body,
+      QueueUrl: process.env.SEARCH_SLA_VIOLATIONS_QUEUE_URL || "queue",
```

Comment:
@FeliceMente io solleverei un'eccezione in caso di variabile di ambiente non definita; quel default è pericoloso.

Reply:
fatto